blocks/vehicle-selector/vehicle-selector.js

```diff
@@ -0,0 +1,241 @@
+/* eslint-disable max-len */
+import { createElement, stripEmptyTags } from '../../scripts/helpers.js';
+import { smoothScrollHorizontal } from '../../scripts/motion-helper.js';
+
+const blockName = 'vehicle-selector';
+
+function buildTabNavigation(tabItems, clickHandler) {
+  const tabNavigation = createElement('ul', { classes: `${blockName}__navigation` });
+  const navigationLine = createElement('li', { classes: `${blockName}__navigation-line` });
+
+  tabItems.forEach((tabItem, i) => {
+    const listItem = createElement('li', { classes: `${blockName}__navigation-item` });
+    const button = createElement('button');
+    button.classList.add('h5');
+    button.addEventListener('click', () => clickHandler(i));
+    // TODO make surer i target the right heading
+    const tabTitle = tabItem.querySelector('h1,h2,h3,h4,h5,h6');
+    button.innerText = tabTitle.innerText;
+    tabTitle.remove();
+
+    listItem.append(button);
+    tabNavigation.append(listItem);
+  });
+
+  tabNavigation.append(navigationLine);
+
+  return tabNavigation;
+}
+
+const updateActiveItem = (index) => {
+  const images = document.querySelector(`.${blockName}__images-container`);
+  const descriptions = document.querySelector(`.${blockName}__description-container`);
+  const navigation = document.querySelector(`.${blockName}__navigation`);
+
+  [images, descriptions, navigation].forEach((c) => c.querySelectorAll('.active').forEach((i) => {
+    i.classList.remove('active');
+
+    // Remove tabindex from previously active items
+    i.querySelectorAll('a').forEach((link) => link.setAttribute('tabindex', '-1'));
+  }));
+
+  images.children[index].classList.add('active');
+  descriptions.children[index].classList.add('active');
+  navigation.children[index].classList.add('active');
+
+  // Make links of current item are accessible by keyboard
+  descriptions.children[index].setAttribute('aria-hidden', 'false');
+  descriptions.children[index].querySelectorAll('a').forEach((link) => link.setAttribute('tabindex', '0'));
+
+  // Center navigation item
+  const navigationActiveItem = navigation.querySelector('.active');
+
+  if (navigation && navigationActiveItem) {
+    const { clientWidth: itemWidth, offsetLeft } = navigationActiveItem;
+    // Calculate the scroll position to center the active item
+    const scrollPosition = offsetLeft - (navigation.clientWidth - itemWidth) / 2;
+    navigation.scrollTo({
+      left: scrollPosition,
+      behavior: 'smooth',
+    });
+  }
+
+  // Update description position
+  const descriptionWidth = descriptions.offsetWidth;
+
+  setTimeout(() => {
+    descriptions.scrollTo({
+      left: descriptionWidth * index,
+      behavior: 'smooth',
+    });
+  }, 50);
```

Comment:
Why do we need the timeout here? The 50ms delay will not be visible to the user. If the `setTimeout` is needed here for another reason, please add a suitable comment in the code.